### ISSUE TITLE
A question about the Instance normalization

### ISSUE THREAD

**Issue by Fairydetail** [sniklaus/pytorch-sepconv#53]
Hello, your latest work "Revisiting Adaptive Convolutions for Video Frame Interpolation" mentions that input normalization can improve the PSNR. So I apply it to my training of softmax-splatting, while it seems to be divergent after several steps. Here is my implemention，is there any problems?  Or the instance normalization is not suitable for the frame interpolation based on optical flow estimation, since the distribution of input data has changed after instance normalization. Looking forward to your reply, thanks a lot.
![image](https://user-images.githubusercontent.com/26640881/100565909-9010c700-32ff-11eb-9a58-cf4a833a2c21.png)
`def instance_norm(I0, I1):
    input = torch.cat((I0, I1), dim=2)
    mean1 = input.mean(axis=(2,3),keepdim=True)
    mean = mean1.expand(input.shape)
    std1 = input.std(axis=(2,3),keepdim=True)
    std = std1.expand(input.shape)
    input_hat = (input-mean)/std
    return input_hat[:, :, :int(input_hat.size(2) / 2), :],\
           input_hat[:, :, int(input_hat.size(2) / 2):, :], mean1, std1

def instance_norm_reverse(result, mean, std):
    std1=std.expand(result.shape)
    mean1=mean.expand(result.shape)
    output = result*std1+mean1
    output[output>1.0] = 1.0
    output[output<0.0] = 0.0
    return output`

**Comment by sniklaus** [comment]
Thank you for your interest in our work! It seems like your code would backprop through the normalization, which it should not. Your code is subject to a division by zero if the image is a solid color. Your code shows the inverse of the normalization but that is not necessary, one can simply apply the adaptive kernels to the non-normalized inputs. The code we use is roughly as follows.

```
tenSeq = [first_frame, second_frame]

with torch.set_grad_enabled(False):
    tenStack = torch.stack(tenSeq, 1)
    tenMean = tenStack.view(tenStack.shape[0], -1).mean(1, True).view(tenStack.shape[0], 1, 1, 1)
    tenStd = tenStack.view(tenStack.shape[0], -1).std(1, True).view(tenStack.shape[0], 1, 1, 1)
    tenSeq = [(tenFrame - tenMean) / (tenStd + 0.0000001) for tenFrame in tenSeq]
    tenSeq = [tenFrame.detach() for tenFrame in tenSeq]
# end
```
